packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,142 @@
+  </button>
+</template>
+
+<style lang="scss">
+.onyx-button {
+  display: inline-flex;
+  padding: var(--onyx-spacing-2xs, 0.5rem) var(--onyx-spacing-sm, 0.75rem);
+  justify-content: center;
+  align-items: center;
+  gap: var(--onyx-spacing-4xs, 0.25rem);
+  border-radius: var(--onyx-radius-sm, 0.25rem);
+  cursor: pointer;
+
+  --background-color: transparent;
+  --background-hover-color: var(--onyx-color-base-primary-100);
+  --background-disabled-color: transparent;
+
+  --color: var(--onyx-color-text-primary-intense);
+  --disabled-color: var(--onyx-color-text-neutral-inverted);
+  --border-color: transparent;
+  --disabled-border-color: transparent;
+  --focus-outline-color: var(--onyx-color-base-primary-200);
+
+  &--secondary {
+    --background-hover-color: var(--onyx-color-base-neutral-200);
+    --focus-outline-color: var(--onyx-color-base-neutral-300);
+    --color: var(--onyx-color-text-neutral-intense);
+    --disabled-color: var(--onyx-color-text-primary-soft);
+  }
+
+  &--danger {
+    --background-hover-color: var(--onyx-color-base-danger-200);
+    --focus-outline-color: var(--onyx-color-base-danger-300);
+    --color: var(--onyx-color-text-danger-intense);
+    --disabled-color: var(--onyx-color-text-danger-medium);
+  }
+
+  &--primary.onyx-button--default {
+    --background-color: var(--onyx-color-base-primary-500);
+    --background-hover-color: var(--onyx-color-base-primary-400);
+    --background-disabled-color: var(--onyx-color-base-primary-200);
+    --color: var(--onyx-color-text-neutral-inverted);
+    --disabled-color: var(--onyx-color-text-neutral-inverted);
+    --border-color: var(--onyx-color-base-primary-500);
+    --disabled-border-color: var(--onyx-color-base-primary-200);
+  }
+
+  &--secondary.onyx-button--default {
+    --background-color: var(--onyx-color-base-background-blank);
+    --background-hover-color: var(--onyx-color-base-neutral-200);
+    --background-disabled-color: var(--onyx-color-base-background-blank);
+    --border-color: var(--onyx-color-base-neutral-400);
+    --disabled-border-color: var(--onyx-color-base-neutral-200);
+  }
+
+  &--danger.onyx-button--default {
+    --background-color: var(--onyx-color-base-danger-200);
+    --background-hover-color: var(--onyx-color-base-danger-100);
+    --background-disabled-color: var(--onyx-color-base-danger-100);
+    --border-color: var(--onyx-color-base-danger-500);
+    --disabled-border-color: var(--onyx-color-base-danger-200);
+  }
+
+  &--primary.onyx-button--outline {
+    --border-color: var(--onyx-color-base-primary-500);
+    --disabled-border-color: var(--onyx-color-base-primary-200);
+  }
+
+  &--secondary.onyx-button--outline {
+    --border-color: var(--onyx-color-base-neutral-400);
+    --disabled-border-color: var(--onyx-color-base-neutral-200);
+  }
+
+  &--danger.onyx-button--outline {
+    --border-color: var(--onyx-color-base-danger-500);
+    --disabled-border-color: var(--onyx-color-base-danger-200);
+  }
```

Comment:
```suggestion
  &--primary {
    &.onyx-button--default {
      --background-color: var(--onyx-color-base-primary-500);
      --background-hover-color: var(--onyx-color-base-primary-400);
      --background-disabled-color: var(--onyx-color-base-primary-200);
      --color: var(--onyx-color-text-neutral-inverted);
      --disabled-color: var(--onyx-color-text-neutral-inverted);
      --border-color: var(--onyx-color-base-primary-500);
      --disabled-border-color: var(--onyx-color-base-primary-200);
    }

    &.onyx-button--outline {
      --border-color: var(--onyx-color-base-primary-500);
      --disabled-border-color: var(--onyx-color-base-primary-200);
    }
  }

  &--secondary {
    --background-hover-color: var(--onyx-color-base-neutral-200);
    --focus-outline-color: var(--onyx-color-base-neutral-300);
    --color: var(--onyx-color-text-neutral-intense);
    --disabled-color: var(--onyx-color-text-primary-soft);

    &.onyx-button--default {
      --background-color: var(--onyx-color-base-background-blank);
      --background-hover-color: var(--onyx-color-base-neutral-200);
      --background-disabled-color: var(--onyx-color-base-background-blank);
      --border-color: var(--onyx-color-base-neutral-400);
      --disabled-border-color: var(--onyx-color-base-neutral-200);
    }

    &.onyx-button--outline {
      --border-color: var(--onyx-color-base-neutral-400);
      --disabled-border-color: var(--onyx-color-base-neutral-200);
    }
  }

  &--danger {
    --background-hover-color: var(--onyx-color-base-danger-200);
    --focus-outline-color: var(--onyx-color-base-danger-300);
    --color: var(--onyx-color-text-danger-intense);
    --disabled-color: var(--onyx-color-text-danger-medium);

    &.onyx-button--default {
      --background-color: var(--onyx-color-base-danger-200);
      --background-hover-color: var(--onyx-color-base-danger-100);
      --background-disabled-color: var(--onyx-color-base-danger-100);
      --border-color: var(--onyx-color-base-danger-500);
      --disabled-border-color: var(--onyx-color-base-danger-200);
    }

    &.onyx-button--outline {
      --border-color: var(--onyx-color-base-danger-500);
      --disabled-border-color: var(--onyx-color-base-danger-200);
    }
  }
```

We could group the classes like so that each color has its own "code block" :) What do you think?